File: tests/fixtures/sample.ts
// Sample TypeScript file for testing ConfigSmith scanner
// This file demonstrates various patterns for accessing environment variables

// Direct access patterns
const dbUrl = process.env.DATABASE_URL || 'postgresql://localhost/db';
const apiKey = process.env.API_KEY;

// Destructuring with defaults
const { PORT = '3000', NODE_ENV = 'development' } = process.env;

// Bracket notation
const secretKey = process.env["SECRET_KEY"];
const debug = process.env["DEBUG"] || 'false';

// Framework patterns (Next.js)
const publicApiUrl = process.env.NEXT_PUBLIC_API_URL;

// React pattern
const reactAppName = process.env.REACT_APP_NAME || 'MyApp';

// Configuration object pattern
const config = {
  database: process.env.DB_HOST || 'localhost',
  port: parseInt(process.env.DB_PORT || '5432', 10),
  ssl: process.env.DB_SSL === 'true',
};

export { dbUrl, apiKey, config };

File: tests/fixtures/sample.py
"""
Sample Python file for testing ConfigSmith scanner.

This file demonstrates various patterns for accessing environment variables.
"""

import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Database configuration
DATABASE_URL = os.getenv("DATABASE_URL", "postgresql://localhost/db")

# API configuration
API_KEY = os.environ.get("API_KEY")  # Required, no default

# Application settings
DEBUG = os.getenv("DEBUG", "false")
PORT = int(os.getenv("PORT", "8000"))  # Type conversion

# Alternative access pattern
SECRET_KEY = os.environ["SECRET_KEY"]

# Cache settings
CACHE_TTL = int(os.environ.get("CACHE_TTL", "300"))
REDIS_URL = os.getenv("REDIS_URL", "redis://localhost:6379")
